api/api.go

```diff
@@ -25,9 +25,9 @@ type PayloadProblems struct {
 	PayloadError
 }
 
-// PayloadSubmission represents metadata about a successful submission.
-type PayloadSubmission struct {
-	*Submission
+// PayloadSubmittedIteration represents metadata about a submission of an iteration.
+type PayloadSubmittedIteration struct {
+	*SubmittedIteration
```

Comment:
I thought the same thing, but there's already an `Iteration` type defined in `iteration.go`. The two types are really similar and could probably be combined with no real consequences, but since that was a much more significant refactor than this, I felt that it would be better to keep two different types rather than trying to merge them now.

After writing all that, though, I'm starting to think that merging those two types might be a good idea. What do you think?